x-pack/plugins/apm/server/routes/settings/anomaly_detection.ts

```diff
@@ -10,14 +10,19 @@ import { getAnomalyDetectionJobs } from '../../lib/anomaly_detection/get_anomaly
 import { createAnomalyDetectionJobs } from '../../lib/anomaly_detection/create_anomaly_detection_jobs';
 import { setupRequest } from '../../lib/helpers/setup_request';
 import { getAllEnvironments } from '../../lib/environments/get_all_environments';
+import { hasLegacyJobs } from '../../lib/anomaly_detection/has_legacy_jobs';
 
 // get ML anomaly detection jobs for each environment
 export const anomalyDetectionJobsRoute = createRoute(() => ({
   method: 'GET',
   path: '/api/apm/settings/anomaly-detection',
   handler: async ({ context, request }) => {
     const setup = await setupRequest(context, request);
-    return await getAnomalyDetectionJobs(setup, context.logger);
+    const jobs = await getAnomalyDetectionJobs(setup, context.logger);
+    return {
+      jobs,
+      hasLegacyJobs: await hasLegacyJobs(setup),
+    };
```

Comment:
`getAnomalyDetectionJobs` and `hasLegacyJobs` can be done in parallel.

Reply:
Good point. I could have sworn I made this change after I saw your comment but I must have forgotten to push or something. 
I'll update in a separate PR.